src/components/Search/Search.tsx

```diff
@@ -1,25 +1,18 @@
 import React from 'react';
```

Comment:
For me the key reason for separating search was debounce logic which should be reused. As for the TextInput, it lacks debounce.

As for this HOC, for me it seems to be useless, as it could be replaced with single useState.

So, I think, new Search should be introduced anyway, even if we leave SettingsSearch untouched.

Reply:
DIscussed this on a call and decided to keep a new search component, but base it on [StorageFilter](https://github.com/ydb-platform/ydb-embedded-ui/tree/main/src/containers/Storage/StorageFilter) instead, which essentially is the same